db/migrate_core_cards/20141230204340_uri_codename.rb

```diff
@@ -0,0 +1,11 @@
+# -*- encoding : utf-8 -*-
+
+class UriCodename < Wagn::Migration
+  def up
+    contentedly do
+      puts "adding new codename card for URI"
+      codename = 'URI'
+      Card.create! :type_id=>Card::CardtypeID, :name=>codename, :codename=>codename.to_name.key
```

Comment:
Yes, maybe.  Shouldn't we handle that in a standard way in migrations?  Seems like you might even want configured options about how to handle name conflicts in the cardname and codename spaces.  We can do like it used to do a long time ago and change the name to URI1 and still use the codename for the original name.  A codename collision is going to be either an error, or re-running the same migration.